portal-impl/src/com/liferay/portal/util/JarUtil.java

```diff
@@ -48,16 +52,28 @@ public static void downloadAndInstallJar(URL url, Path path)
 			Files.copy(inputStream, path, StandardCopyOption.REPLACE_EXISTING);
 		}
 
+		try (InputStream inputStream = Files.newInputStream(path)) {
+			String digest = DigesterUtil.digestHex(Digester.SHA_1, inputStream);
+
+			if (!Objects.equals(sha1, digest)) {
```

Comment:
@dantewang You may want to normalize the two hashes here by doing a toUpperCase or toLowerCase since hashes are commonly written in uppercase and in lowercase.

Reply:
@samuelkong Switched to `StringUtil.equalsIgnoreCase(String, String)`, see https://github.com/liferay-core-infra/liferay-portal/pull/89/commits/1ad1930b6ad7a1df213a73fdcb8d06b8ce72cbed. @tinatian